### ISSUE TITLE
Add Contributor License Agreements (CLA)

### ISSUE THREAD

**Issue by b-g** [hfg-gmuend/openmoji#120]
cc @carlinmack @kriskowal 

This is to discuss:

> ... in terms of copyright I think by contributing you agree to release under CC BY SA 4.0, however I would love to have a conversation about this topic

https://github.com/hfg-gmuend/openmoji/pull/118#issuecomment-576987491

IMO we should mention that and somehow make it part of the PR pipeline that people are aware of it and agree to it. I guess implicit this is already the case ... but something more "formal" would be great!

Any best practice examples how other more mature projects are dealing with it?

**Comment by kriskowal** [comment]
I have had good experiences with CLA Assistant, which is a Github extension that you can add as a blocking reviewer, pending an explicit copyright assignment.

https://cla-assistant.io/

**Comment by Daniel-Mietchen** [comment]
Yes, CLA Assistant would be my recommendation as well. In use, for instance, at Grafana, so make a PR there, and you can see how this goes. Simple example: https://github.com/grafana/grafana/pull/20817 .

**Comment by b-g** [comment]
I don't think we will need formally a CLA as this seems to be covered with the general terms of use for every Github user:

https://help.github.com/en/github/site-policy/github-terms-of-service#6-contributions-under-repository-license

> Whenever you make a contribution to a repository containing notice of a license, you license your contribution under the same terms, and you agree that you have the right to license your contribution under those terms. If you have a separate agreement to license your contributions under different terms, such as a contributor license agreement, that agreement will supersede.
> 
> Isn't this just how it works already? Yep. This is widely accepted as the norm in the open-source community; it's commonly referred to by the shorthand "inbound=outbound". We're just making it explicit.

And then I agree with this article "[Why you probably shouldn’t add a CLA to your open source project](https://ben.balter.com/2018/01/02/why-you-probably-shouldnt-add-a-cla-to-your-open-source-project/#if-a-license-isnt-good-enough-for-maintainers-you-shouldnt-subject-users-to-it)" especially:

>  CLAs require that the first interaction between an open source project and a potential contributor to involve a formal and complex legal agreement which signs away their legal rights — not exactly a warm welcome. Submitting one’s first contribution to an open source project is already scary enough as it is, even without the threat of a lawsuit as the first volley.

However I think we should make "inbound=outbound" explicit in the FAQ and CONTRIBUTING.md

Opinions?

**Comment by carlinmack** [comment]
Another topic I'd like discussed is, is there room for someone to release their emoji under a less restrictive license? I.e. in `openmoji.csv` there could be a field that is by default `CC BY-SA 4.0` and could be CC0 for example. 

Alternatively is it okay to release your design on another platform and include it in OpenMoji - have the same file under two licenses when distributed on different platforms?

**Comment by kriskowal** [comment]
Well, I am not a lawyer and what follows is not legal advice.

My understanding is that without a CLA, the originating author retains the copyright and thus the right to grant other licenses.

My interpretation of the Github terms of use is that contributors to this project implicitly grant their work under the same terms as the project’s posted license (at the time of their contribution!), and retain the copyright for their work.

With a CLA, the project receiving the copyright gains the ability to grant other licenses. For example, without a CLA, the holders of the copyright do not have the right to relax or constrain the license.

I do not have a preference for what the copyright holders of this project decide to do.

**Comment by b-g** [comment]
@kriskowal 
> Well, I am not a lawyer and what follows is not legal advice. 

++ Same here :)

> My understanding is that without a CLA, the originating author retains the copyright and thus the right to grant other licenses.

> My interpretation of the Github terms of use is that contributors to this project implicitly grant their work under the same terms as the project’s posted license (at the time of their contribution!), and retain the copyright for their work.

Same interpretation here, but I think this is how it should be. 

s/he will already have granted openmoji the right to use it. Hence will no longer be able to grant an exclusive license to others and is only to be able to grant licenses for the individual emojis s/he contributed e.g. in your case the Space Rocket. So you still can sell your Space Rocket or do whatever you want with it ... but grant an exclusive license to others (and prevent OpenMoji using it under CC BY-SA 4.0)

Also IMO the risk is really low for the entire project, as s/he is not allowed to grant licenses in the name of the entire OpenMoji project (think of brand) or emojis of other contributors.

Basically I think it is fine as it is ... but we should mention it better.

@carlinmack 
I think this answers your question as well. You can do all that with your contributions ... But we should keep it simple and stick with a single outbound license: CC BY-SA 4.0

OK?

**Comment by b-g** [comment]
Addressed and IMO reflected as disccused with https://github.com/hfg-gmuend/openmoji/commit/bd1705c82d650c74b0cd1ab2c24a47d56b3caa88

Please re-open if I missed something. Many thanks for all the very helpful arguments!
